### DISCUSSION TITLE
Google Geospatial API

### DISCUSSION THREAD

**Discussion by raymondskl** [sceneview/sceneview#332] [Q&A]
Just a quick question, as a search didn't return any answers. Does Sceneview currently support Streetscape Geometry and Terrain Anchors?

**Comment by ThomasGorisse** [comment]
SceneView's `ARSceneView` in v4.0.0-rc.1 gives you access to the ARCore `Session` and `Frame`, which you can use with the Geospatial API:

```kotlin
ARSceneView(
    sessionConfiguration = { session, config ->
        config.geospatialMode = Config.GeospatialMode.ENABLED
    },
    onSessionUpdated = { session, frame ->
        val earth = session.earth
        if (earth?.trackingState == TrackingState.TRACKING) {
            val pose = earth.cameraGeospatialPose
            // Use lat/lng/altitude for geospatial anchors
        }
    }
)
```

The Geospatial API is an ARCore feature — SceneView provides the session/frame access, and you use Google's Geospatial APIs directly.
